src/client/Client.cc

```diff
@@ -11584,8 +11584,12 @@ int Client::_do_setxattr(Inode *in, const char *name, const void *value,
 {
 
   int xattr_flags = 0;
-  if (!value)
-    xattr_flags |= CEPH_XATTR_REMOVE;
+  // If the size is set to 0, set the value to an empty string
+  if (size == 0) {
+    value = "";
+  } else if (value == NULL) {
+      return -EINVAL;
```

Comment:
A simple fix is just adding this code at very beginning of Client::_setxattr, instead here

Reply:
Thanks @ukernel. Yea I see some stat mismatches in the tests. I think this may fix it.

Reply:
I mean just this simple change, remove all other changes

Reply:
Ack have refactored it. Thanks @ukernel!